Source/BugsnagCrashReport.m

```diff
@@ -620,6 +645,42 @@ - (NSString *_Nullable)enhancedErrorMessageForThread:(NSDictionary *_Nullable)th
 
 @end
 
+@implementation FallbackReportData
+
+- (instancetype)initWithMetadata:(NSString *)metadata {
```

Comment:
It'd be nice to separate this out into individual methods to improve readability (e.g. one that returns the severity, another that returns the handled state, etc).